op-node/rollup/interop/managed/system.go

```diff
@@ -353,10 +352,30 @@ func (m *ManagedMode) Reset(ctx context.Context, lUnsafe, xUnsafe, lSafe, xSafe,
 		logger.Error("Cannot reset, cross-safe block not known")
 		return err
 	}
-	finalizedRef, err := verify(finalized, "finalized")
-	if err != nil {
+
+	var finalizedRef eth.L2BlockRef
+
+	if finalized == (eth.BlockID{}) {
+		logger.Debug("No finalized reset target provided, using local finalized block")
+		ref, err := m.l2.L2BlockRefByLabel(ctx, eth.Finalized)
+		if err != nil {
+			logger.Error("Cannot reset, finalized block not known locally")
+			return &gethrpc.JsonError{
+				Code:    InternalErrorRPCErrcode,
+				Message: "failed to find finalized reference",
+			}
+		}
+		if ref.Number > xSafe.Number {
+			logger.Warn("Finalized block is newer than cross-safe, using cross-safe as finalized reset target instead")
+			finalizedRef = xSafeRef
+		} else {
+			finalizedRef = ref
+		}
+	} else if ref, err := verify(finalized, "finalized"); err != nil {
 		logger.Error("Cannot reset, finalized block not known")
 		return err
+	} else {
+		finalizedRef = ref
 	}
```

Comment:
This is a hack currently to make resets work. Since pre-Interop the node still manages its own finalized L2s, this works, but it may be cleaner to also let the supervisor already handle L2 finalization pre-Interop.

Reply:
Ideally we just fix supervisor to handle finalized.